fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/MergeProjects.java

```diff
@@ -92,7 +95,7 @@ public Expression visit(Expression expr, Map<Expression, Expression> substitutio
                 Slot ref = ((SlotReference) expr).withQualifier(Collections.emptyList());
                 if (substitutionMap.containsKey(ref)) {
                     Alias res = (Alias) substitutionMap.get(ref);
-                    return res.child();
+                    return new Alias(res.child(), ref.getName());
```

Comment:
the new Alias is useless and will intro more complexity to expression process. because when we come here the slot is not the root of this expr tree.

Reply:
fixed